components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationMgtUtil.java

```diff
@@ -1212,4 +1214,24 @@ public void serialize(InboundAuthenticationRequestConfig value, JsonGenerator ge
             gen.writeEndObject();
         }
     }
+
+    /**
+     * Check whether the trusted app consent is granted for the application.
+     *
+     * @param serviceProvider Service provider.
+     * @return True if 'trustedAppConsentGranted' spProperty of the application is true.
+     */
+    public static boolean isTrustedAppConsentGranted(ServiceProvider serviceProvider) {
+
+        if (serviceProvider != null) {
+            String trustedAppConsent = Arrays.stream(serviceProvider.getSpProperties())
+                    .filter(spProp -> StringUtils.equals(spProp.getName(),
+                            TRUSTED_APP_CONSENT_GRANTED_SP_PROPERTY_NAME))
+                    .map(ServiceProviderProperty::getValue)
+                    .findFirst()
+                    .orElse(null);
```

Comment:
Lets extract this to a private method for getting the prop value

Reply:
This method will be removed as the consent granted attribute will be available as an attribute of the trustedAppMetadata object.